### ISSUE TITLE
Minetest crashes upon using /spawnentity

### ISSUE THREAD

**Issue by C1ffisme** [minetest/minetest#3412]
```
minetest: /build/minetest-Si3yyz/minetest-201511241831/src/util/serialize.h:268: void writeF1000(irr::u8*, irr::f32): Assertion `i >= ((float)(s32)((-0x7FFFFFFF - 1) / 1000.0f)) && i <= ((float)(s32)((0x7FFFFFFF) / 1000.0f))' failed.
Aborted (core dumped)
```

This might be a duplicate. I think it looks familiar.

EDIT:
It may or may not be a problem with my mod, but even then, minetest shouldn't crash like this.

**Comment by C1ffisme** [comment]
EDIT:
Yes, I recognize this. It happens when you set an entity's yaw to NaN. I think this is worth a fix in the C++ code, though. At least it should give an error message.

**Comment by est31** [comment]
#3326 will fix this.

**Comment by C1ffisme** [comment]
That's good. At least I gained knowledge.

**Comment by 0-afflatus** [comment]
This might explain the problems I was having with a mob spawner using minetest.add_entity()?

**Comment by C1ffisme** [comment]
Yeah, it happens with me when I'm setting the yaw of an entity to face a player. If the player is in the same place as an entity, then you generate an NaN.

**Comment by tenplus1** [comment]
If you were spawning an entity from Mobs Redo then this error has been fixed already, you can close this issue :)

**Comment by 0-afflatus** [comment]
@tenplus1 the issue in mobs_redo is fixed, sure, but this should stay open until #3326 goes through.

**Comment by nerzhul** [comment]
This should have been fixed in minetest 5.0
